### ISSUE TITLE
Wrong reported operand size on cmp and cmpsb

### ISSUE THREAD

**Issue by EliaGeretto** [quarkslab/QBDI#57]
When using instruction analysis, the reported size of the operands for the following instructions is wrong:
* `cmp eax, imm32` reports 8 and 32, but should be 4 and 4.
* `cmpsb byte ptr [rsi], byte ptr es:[rdi]` reports 32 and 32, but should be 1 and 1.

This is likely caused by an LLVM bug.

**Comment by haxelion** [comment]
Yes I suspect as well that this is an incorrect information in LLVM tables.
This will require more time to be fixed.

**Comment by nezetic** [comment]
All this was due to various bugs in operand analysis, not discovered because of the very limited regression tests. #57 aims at fixing the issues.
